File: 10-greedy/06_non_overlapping.c
#include "../test_runner.h"

int erase_overlap_intervals(int intervals[][2], int n) {
    return -1;
}

int main() {
    int i1[][2] = {{1,2},{2,3},{3,4},{1,3}};
    TEST_INT("one remove", erase_overlap_intervals(i1, 4), 1);
    int i2[][2] = {{1,2},{1,2},{1,2}};
    TEST_INT("two remove", erase_overlap_intervals(i2, 3), 2);
    int i3[][2] = {{1,2},{2,3}};
    TEST_INT("no remove",  erase_overlap_intervals(i3, 2), 0);
    TEST_SUMMARY();
}

File: 10-greedy/06_non_overlapping.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Non-overlapping Intervals (LC #435) -- Medium
# Min intervals to remove to make rest non-overlapping.
#
#   Key: Sort by end time. Greedily keep intervals that don't overlap with last kept.
#   Equivalent to: Total intervals - max non-overlapping intervals (activity selection)

def erase_overlap_intervals(intervals):
    """Return min intervals to remove"""
    pass


run_tests(erase_overlap_intervals, [
    (([[1,2],[2,3],[3,4],[1,3]],), 1),
    (([[1,2],[1,2],[1,2]],),       2),
    (([[1,2],[2,3]],),             0),
])
